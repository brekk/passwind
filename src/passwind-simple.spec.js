import path from 'path'
// import { pipe, map } from 'ramda'
import { fork } from 'fluture'
// import {hook} from 'ripjam/test'
import { passwind } from './index'
const local = x => path.join(__dirname, x)

const fixture = {
  css: local('../fixture/example.css'),
  smallCSS: local('../fixture/small-example.css'),
  html: local('../fixture/example.html'),
}

test('passwind - small case', done => {
  fork(done)(result => {
    expect(result).toMatchSnapshot()
    done()
  })(passwind({}, fixture.smallCSS, fixture.html))
})
