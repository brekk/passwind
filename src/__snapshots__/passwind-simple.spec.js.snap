// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passwind - small case 1`] = `
Array [
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".min-h-screen": Object {
        "minHeight": "100vh",
      },
    },
    "id": "gspemxyf",
    "selector": Array [
      "min-h-screen",
      "flex",
      "flex-col",
    ],
  },
  Object {
    "definitions": Object {
      ".hover:bg-white": Object {
        "--tw-bg-opacity": "1",
        "backgroundColor": "rgba(255, 255, 255, var(--tw-bg-opacity))",
      },
    },
    "id": "mcmgofus",
    "selector": Array [
      "font-extended",
      "relative",
      "hover:bg-white",
      "hover:shadow-navbar",
      "hover:text-black",
      "bg-ifpink",
      "text-black",
    ],
  },
  Object {
    "definitions": Object {
      ".h-screen": Object {
        "height": "100vh",
      },
    },
    "id": "fhxblwav",
    "selector": Array [
      "absolute",
      "z-20",
      "h-screen",
      "w-screen",
      "bg-white",
      "text-black",
      "font-extended",
      "transition-all",
      "transform-gpu",
      "-translate-x-full",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
    },
    "id": "yajtwnup",
    "selector": Array [
      "flex",
      "flex-col",
      "px-5",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
    },
    "id": "gurttxgm",
    "selector": Array [
      "flex",
      "my-10",
      "justify-between",
      "items-center",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".h-full": Object {
        "height": "100%",
      },
    },
    "id": "elfkwtca",
    "selector": Array [
      "flex",
      "items-center",
      "h-full",
      "relative",
    ],
  },
  Object {
    "definitions": Object {
      ".ml-2": Object {
        "marginLeft": "0.5rem",
      },
    },
    "id": "cddyexlt",
    "selector": Array [
      "ml-2",
      "text-black",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".h-10": Object {
        "height": "2.5rem",
      },
      ".h-12": Object {
        "height": "3rem",
      },
      ".hover:bg-black": Object {
        "--tw-bg-opacity": "1",
        "backgroundColor": "rgba(0, 0, 0, var(--tw-bg-opacity))",
      },
      ".ml-4": Object {
        "marginLeft": "1rem",
      },
    },
    "id": "tarvliqr",
    "selector": Array [
      "flex",
      "justify-center",
      "items-center",
      "font-extended",
      "rounded-full",
      "whitespace-nowrap",
      "transition",
      "border-black",
      "border-2",
      "bg-transparent",
      "text-black",
      "hover:bg-black",
      "hover:text-white",
      "p-4",
      "h-10",
      "h-12",
      "ml-4",
      "py-3",
      "px-6",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
    },
    "id": "animviej",
    "selector": Array [
      "flex",
      "items-stretch",
      "justify-between",
      "px-3",
      "lg:px-16",
    ],
  },
  Object {
    "definitions": Object {
      ".hidden": Object {
        "display": "none",
      },
      ".md:flex": Object {
        "display": "flex",
      },
    },
    "id": "wvrxbkiw",
    "selector": Array [
      "hidden",
      "md:flex",
      "items-center",
      "lg:text-xl",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".h-full": Object {
        "height": "100%",
      },
    },
    "id": "tobmwfjl",
    "selector": Array [
      "px-3",
      "lg:px-4.5",
      "h-full",
      "flex",
      "items-center",
      "whitespace-nowrap",
    ],
  },
  Object {
    "definitions": Object {
      ".md:hidden": Object {
        "display": "none",
      },
    },
    "id": "cydfagzb",
    "selector": Array [
      "md:hidden",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".h-auto": Object {
        "height": "auto",
      },
      ".mb-16": Object {
        "marginBottom": "4rem",
      },
      ".md:px-4": Object {
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
      },
    },
    "id": "yiitxqor",
    "selector": Array [
      "offset-box",
      "z-10",
      "w-4/5",
      "flex",
      "flex-col",
      "m-auto",
      "md:px-4",
      "h-auto",
      "mb-16",
      "border-opacity-100",
      "border-2",
      "border-solid",
      "border-black",
      "bg-white",
      "items-center",
      "mt-8",
      "px-5",
    ],
  },
  Object {
    "definitions": Object {
      ".mb-8": Object {
        "marginBottom": "2rem",
      },
    },
    "id": "ygkkcqrk",
    "selector": Array [
      "text-4xl",
      "text-center",
      "mb-8",
      "mt-16",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".h-16": Object {
        "height": "4rem",
      },
      ".max-h-16": Object {
        "maxHeight": "4rem",
      },
    },
    "id": "njodlcss",
    "selector": Array [
      "font-favorit",
      "flex",
      "flex-col",
      "px-2",
      "py-3",
      "mt-3",
      "w-full",
      "border",
      "rounded-plus",
      "border-solid",
      "h-16",
      "max-h-16",
      "border-black",
    ],
  },
  Object {
    "definitions": Object {
      ".mb-px3": Object {
        "marginBottom": "0.1875rem",
      },
    },
    "id": "kqgdeile",
    "selector": Array [
      "text-xs",
      "mb-px3",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
    },
    "id": "iuzutywi",
    "selector": Array [
      "flex",
    ],
  },
  Object {
    "definitions": Object {
      ".mb-8": Object {
        "marginBottom": "2rem",
      },
    },
    "id": "almvmixl",
    "selector": Array [
      "p-2",
      "text-sm",
      "mb-8",
      "mt-2",
      "bg-alertyellow",
      "text-center",
      "w-full",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".hover:bg-transparent": Object {
        "backgroundColor": "transparent",
      },
      ".mb-4": Object {
        "marginBottom": "1rem",
      },
      ".md:px-4": Object {
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
      },
      ".md:py-5": Object {
        "paddingBottom": "1.25rem",
        "paddingTop": "1.25rem",
      },
      ".md:text-xl": Object {
        "fontSize": "1.25rem",
        "lineHeight": "1.75rem",
      },
    },
    "id": "oiblughi",
    "selector": Array [
      "flex",
      "justify-center",
      "items-center",
      "font-extended",
      "rounded-full",
      "whitespace-nowrap",
      "transition",
      "border-black",
      "border-2",
      "bg-black",
      "text-white",
      "hover:bg-transparent",
      "hover:text-black",
      "w-11/12",
      "sm:w-7/12",
      "mb-4",
      "text-lg",
      "md:text-xl",
      "p-3",
      "md:py-5",
      "md:px-4",
    ],
  },
  Object {
    "definitions": Object {
      ".mb-4": Object {
        "marginBottom": "1rem",
      },
    },
    "id": "ebhnjsmz",
    "selector": Array [
      "font-favorit",
      "p-2",
      "w-11/12",
      "sm:w-7/12",
      "mb-4",
      "text-xs",
      "text-center",
    ],
  },
  Object {
    "definitions": Object {
      ".md:px-12": Object {
        "paddingLeft": "3rem",
        "paddingRight": "3rem",
      },
    },
    "id": "mkglgibu",
    "selector": Array [
      "px-0",
      "md:px-12",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".md:flex-row": Object {
        "flexDirection": "row",
      },
      ".md:w-full": Object {
        "width": "100%",
      },
    },
    "id": "oiourdzo",
    "selector": Array [
      "flex",
      "flex-col",
      "md:flex-row",
      "w-52",
      "md:w-full",
      "m-auto",
      "justify-between",
      "py-14",
    ],
  },
  Object {
    "definitions": Object {
      ".md:mt-0": Object {
        "marginTop": "0",
      },
    },
    "id": "fuhfdzhy",
    "selector": Array [
      "mt-8",
      "md:mt-0",
      "mx-1",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
      ".h-10": Object {
        "height": "2.5rem",
      },
      ".hover:bg-transparent": Object {
        "backgroundColor": "transparent",
      },
      ".md:mt-0": Object {
        "marginTop": "0",
      },
    },
    "id": "otxutyst",
    "selector": Array [
      "flex",
      "justify-center",
      "items-center",
      "font-extended",
      "rounded-full",
      "whitespace-nowrap",
      "transition",
      "border-black",
      "border-2",
      "bg-black",
      "text-white",
      "hover:bg-transparent",
      "hover:text-black",
      "p-4",
      "h-10",
      "mt-8",
      "md:mt-0",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
    },
    "id": "zesnvftt",
    "selector": Array [
      "flex",
      "justify-between",
      "items-center",
      "border-t",
      "w-full",
      "p-4",
    ],
  },
  Object {
    "definitions": Object {
      ".hidden": Object {
        "display": "none",
      },
      ".md:inline": Object {
        "display": "inline",
      },
    },
    "id": "mhezyybm",
    "selector": Array [
      "hidden",
      "md:inline",
    ],
  },
  Object {
    "definitions": Object {
      ".flex": Object {
        "display": "flex",
      },
    },
    "id": "pmdwmqji",
    "selector": Array [
      "flex",
      "justify-between",
      "items-center",
      "w-40",
    ],
  },
]
`;
